profiler/src/main/java/com/splunk/opentelemetry/profiler/EventProcessingChain.java

```diff
@@ -74,15 +77,39 @@ void accept(RecordedEvent event) {
    * the buffer. After flushing, the buffer will be empty.
    */
   public void flushBuffer() {
-    buffer.stream()
-        .sorted(Comparator.comparing(RecordedEvent::getStartTime))
-        .forEach(this::dispatchEvent);
+    buffer.sort(Comparator.comparing(RecordedEvent::getStartTime));
+    updateAllocationSampler();
+
+    buffer.stream().forEach(this::dispatchEvent);
```

Comment:
No need for `.stream()`
```suggestion
    buffer.forEach(this::dispatchEvent);
```